tools/Dashboard/DevHome.Dashboard/Services/WidgetHostingService.cs

```diff
@@ -2,19 +2,181 @@
 // Licensed under the MIT license.
 
 using System;
+using System.Linq;
 using System.Threading.Tasks;
-using DevHome.Dashboard.Helpers;
+using DevHome.Common.Helpers;
+using DevHome.Common.Services;
 using Microsoft.Windows.Widgets.Hosts;
+using Windows.ApplicationModel.Store.Preview.InstallControl;
+using Log = DevHome.Dashboard.Helpers.Log;
 
 namespace DevHome.Dashboard.Services;
 
 public class WidgetHostingService : IWidgetHostingService
 {
+    private readonly IPackageDeploymentService _packageDeploymentService;
+
+    private static readonly string WidgetServiceStorePackageId = "9N3RK8ZV2ZR8";
+    private const int StoreInstallTimeout = 60_000;

```

Comment:
Didn't know an underscore is a valid character for an integer as a separator, handy!

Recommend changing this to a Timespan, which an overload of Task.Delay accepts.